File: Data Structures & Algorithms/two-integer-sum/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> numeros = new Dictionary<int,int>();
        for(int i = 0; i < nums.Length; i++){
            if(numeros.ContainsKey(target - nums[i])){
                return [numeros[target-nums[i]],i];
            }
            numeros.Add(nums[i],i);
        }
        return[];
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-2.cpp
class Solution {
   public:
    void combinationSumBuilder(vector<int>& nums, vector<int>& subset, int& target,
                               vector<vector<int>>& subsetContainer, int index, int sum) {
        if (sum >= target) {
            if (sum == target) subsetContainer.push_back(subset);
            return;
        }

        for (int i = index; i < nums.size(); i++) {
            subset.push_back(nums[i]);
            combinationSumBuilder(nums, subset, target, subsetContainer, i, sum + nums[i]);
            subset.pop_back();
        }
    }
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<vector<int>> subsets;
        vector<int> subset;
        combinationSumBuilder(nums,subset, target, subsets, 0, 0);
        return subsets;
    }
};
